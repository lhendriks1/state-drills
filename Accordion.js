import React, { Component } from 'react';
import './Accordion.css';

export default class Accordion extends React.Component {
  static defaultProps = {
    sections: []
  };

  state = {
    currentIndex: null
  };

  handleClick(index) {
    this.setState({currentIndex: index});
  };


  renderItems() {
    const {currentIndex} = this.state;
    const {sections} = this.props;

    if (!!sections.length === 0 || !!sections === undefined) {
      return <li></li>
    }
    else {
      return (
      sections.map((section, index) => (
        <li className='Accordion__item' key={index}>
          <button onClick={() => this.handleClick(index)}>
          {section.title}
          </button>
          {currentIndex === index && <p>{section.content}</p>}
        </li>
      ))
    )}
  };

  render() {
    return (
    <div>
      <ul className='Accordion'>
        {this.renderItems()}
      </ul>
    </div>
  )}
}
