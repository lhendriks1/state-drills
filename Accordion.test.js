import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordion from './Accordion';

{ /* sections object for snapshot tests */}

const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
];

{ /* Accordion tests suite */}
describe('Accordion component', () => {

  { /* smoke test */ }
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  { /* jest snapshot tests */ }
  it('renders an empty <li /> given no sections', () => {
    const wrapper = shallow(<Accordion />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders no section as active by default', () => {
    const wrapper = shallow(<Accordion sections={sections} />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('opens any clicked section', () => {
    const wrapper = shallow(<Accordion sections ={sections} />);
    wrapper.find('button').at(1).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('only opens one section at a time', () => {
    const wrapper = shallow(<Accordion sections={sections} />);
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(1).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
