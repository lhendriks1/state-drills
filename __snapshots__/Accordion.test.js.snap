// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component only opens one section at a time 1`] = `
<div>
  <ul
    className="Accordion"
  >
    <li
      className="Accordion__item"
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      className="Accordion__item"
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
      <p>
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      className="Accordion__item"
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`Accordion component opens any clicked section 1`] = `
<div>
  <ul
    className="Accordion"
  >
    <li
      className="Accordion__item"
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      className="Accordion__item"
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
      <p>
        Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
      </p>
    </li>
    <li
      className="Accordion__item"
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`Accordion component renders an empty <li /> given no sections 1`] = `
<div>
  <ul
    className="Accordion"
  />
</div>
`;

exports[`Accordion component renders no section as active by default 1`] = `
<div>
  <ul
    className="Accordion"
  >
    <li
      className="Accordion__item"
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      className="Accordion__item"
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      className="Accordion__item"
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;
